### PR TITLE
update node-datachannel, make WebRTC endpoint unit tests more stable

### PR DESCRIPTION
**Pull request by juslesan**
- General stability updates for the WebRTC endpoint
- WebRTC endpoint tests fast reconnect tests much less flaky on local Ubuntu 20 VM (More instability in CI)

### CODE REVIEW

File: src/connection/Connection.ts [hunk 1]
```diff
@@ -443,12 +446,11 @@ export class Connection extends ConnectionEmitter {
         this.paused = false
         this.dataChannelEmitter = DataChannelEmitter(dataChannel)
         dataChannel.setBufferedAmountLowThreshold(this.bufferThresholdLow)
-        if (this.isOffering) {
-            this.dataChannelEmitter.on('open', () => {
-                this.logger.debug('dc.onOpen')
-                this.openDataChannel(dataChannel)
-            })
-        }
+        this.dataChannelEmitter.on('open', () => {
```
Comment (harbu) [inline src/connection/Connection.ts:449]: Is this if the other side ends up offering after all?

**Review by harbu — APPROVED** [review]

File: src/connection/Connection.ts (hunk 1 above)
Comment (juslesan) [reply src/connection/Connection.ts:449]: This is a reaction to a change in the libdatachannel API. In the WebRTC standard both ends should be able to call onOpen regardless of who is offering. This was not possible previously.

File: src/connection/Connection.ts [hunk 2]
```diff
@@ -301,6 +301,9 @@ export class Connection extends ConnectionEmitter {
             }
         }
 
+        if (this.flushRef) {
+            clearImmediate(this.flushRef)
+        }
```
Comment (timoxley) [inline src/connection/Connection.ts:306]: Good catch

**Review by timoxley — APPROVED** [review]
Nice one.

@juslesan do you know or have any theory as to exactly why this improves stability

**Comment by juslesan** [comment]
> Nice one.
> 
> @juslesan do you know or have any theory as to exactly why this improves stability

@timoxley 

Obviously there's been a punch of updates in libdatachannel between 0.1.2 and 0.1.3 of node-datachannel. It appeared that the never version was stabler in local VM. Also, the onDatachannel() event handler being available for both ends seems to have been an improvement. After using the handler on both ends the tests passed at much higher rates on local VM. Atleast the fast reconnect test case seems to be much less flaky than before.

Currently I'm 99% certain that the flakiness of test case 'messages are delivered on temporary loss of connectivity' is timing related and the issue is likely caused by the underlying libraries. After adding a console.log() for debugging before each send call the test was no longer flaky. Maybe there needs to be a timeout before flushing messages on a new datachannel. However, my most recent attempt at it did not fix the problem in CI.

**Pull request merged**